editoast/src/modelsv2/connection_pool/db_connection_pool.rs

```diff
@@ -0,0 +1,85 @@
+use diesel_async::pooled_connection::deadpool::Object;
+use diesel_async::pooled_connection::deadpool::Pool;
+#[cfg(test)]
+use diesel_async::pooled_connection::AsyncDieselConnectionManager;
+#[cfg(test)]
+use diesel_async::AsyncConnection;
+use diesel_async::AsyncPgConnection;
+use std::sync::Arc;
+#[cfg(test)]
+use tokio::sync::OwnedRwLockWriteGuard;
+#[cfg(test)]
+use tokio::sync::RwLock;
+#[cfg(test)]
+use url::Url;
+
+use super::DbConnectionError;
+
+#[derive(Clone)]
+pub struct DbConnectionPoolV2 {
+    pub pool: Arc<Pool<AsyncPgConnection>>,
+    #[cfg(test)]
+    pub test_connection: Option<Arc<RwLock<Object<AsyncPgConnection>>>>,
+}
+
+impl DbConnectionPoolV2 {
+    pub fn new(pool: Pool<AsyncPgConnection>) -> Self {
```

Comment:
Wouldn't it make sense that the `new` function takes a URL instead of a Pool? Because it looks weird that for building a pool (`DbConnectionPoolV2`), you need a pool (`Pool<AsyncPgConnection>`). It seems that the API is maybe not at the right level.